jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy

```diff
@@ -352,6 +357,7 @@ def run(params) {
                             echo 'Register monitoring server as minion with gui'
                             res_init_monitoring = sh(script: "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd '${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_init_monitoring'", returnStatus: true)
                             echo "Init Monitoring Server status code: ${res_init_monitoring}"
+                            sh "exit ${res_init_monitoring}"
```

Comment:
I would only do this change for proxy, not for monitoring. We can keep testing without monitoring.

Reply:
To my understanding:

- We have a try/finally wrapping the whole pipeline

- Proxy stages are not wrapped in a try/catch at the stages level, a failure in one of them should make the pipeline skip ahead to the `finally` block.

 - Monitoring stages  are wrapped in a try/catch, so a fail here should keep the pipeline running to the end and mark it as red only once done and inside the `finally ` block.

Reply:
Actually, I see your point too.

From sh docs 

> Normally, a script which exits with a nonzero status code will cause the step to fail with an exception

and I think this should be true also for calling `sh exit` with a code != 0
But I'm not 100% sure about it.